examples/systemcalls/systemcalls.c

```diff
@@ -80,20 +118,53 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
-
 
 /*
- * TODO
  *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
  *   redirect standard out to a file specified by outputfile.
  *   The rest of the behaviour is same as do_exec()
  *
 */
 
+	int kidpid;
+	int status;
+	int fd = open(outputfile, O_WRONLY|O_TRUNC|O_CREAT, 0644);
+	if (fd < 0) { perror("open"); abort(); }
+	
+	switch (kidpid = fork()) 
+	{
+	case -1: 
+		perror("fork"); 
+		return false;
```

Comment:
@PeterBraganza in most of the return false cases you are missing cleanup of va_end.  Suggest considering a design where you set a variable and return in one place after va_end